src/IonEvent.ts

```diff
@@ -404,12 +404,17 @@ class IonStructEvent extends AbsIonContainerEvent {//no embed support as of yet.
     structsEqual(actualEvents : AbstractIonEvent[], expectedEvents : AbstractIonEvent[]) : boolean {
         let matchFound : boolean = true;
         let paired : boolean[] = new Array<boolean>(expectedEvents.length);
-        for(let i : number = 0; matchFound && i < actualEvents.length; i++) {
+        for (let i : number = 0; matchFound && i < actualEvents.length; i++) {
             matchFound = false;
-            for(let j : number = 0; !matchFound && j < expectedEvents.length; j++) {
-                if(!paired[j]) {
+            for (let j : number = 0; !matchFound && j < expectedEvents.length; j++) {
+                if (!paired[j]) {
                     matchFound = actualEvents[i].equals(expectedEvents[j]);
-                    paired[j] = matchFound;
+                    if (matchFound && actualEvents[i].eventType === IonEventType.CONTAINER_START) {
+                        for (let k = j; k < expectedEvents[j].ionValue.length; k++) {
+                            paired[k] = true;
+                        }
+                        i += actualEvents[i].ionValue.length;
+                    }
                 }
             }
         }
```

Comment:
if `expectedEvents` matches all the events of `actualEvents`, but `expectedEvents` contains additional, unmatched events, i believe this method will incorrectly return `true`

Reply:
this.structsEqual(this.ionValue, expected.ionValue) && this.structsEqual(expected.ionValue, this.ionValue);

Reply:
> * please try removing good/equivs/lists.ion and good/equivs/sexps.ion from the skipList and see if TravisCI no longer timesout (see the toSkipList() method in iontests.js)
> * for the value `{a: 1, b: {c: 3}, d: 4}`, event stream has the following structure:
> 
> ```
>   SCALAR 1 int a: 1
>   CONTAINER_START 1 struct b: 
>     SCALAR 2 int c: 3
>     CONTAINER_END 1
>   SCALAR 2 int c: 3   // should the event for nested field 'c' appear here as well as above?
>   CONTAINER_END 1
>   SCALAR 1 int d: 4
>   CONTAINER_END 0
> ```
> 
> is this the expected structure?

a container start event indexes all events within (not itself) as well as its container end event.

Reply:
I removed the test vectors from the skip list and the build is not timing out.

Reply:
// different field/value pairs
[
    { a:4, a:4, b:6 },
    { a:4, b:6, b:6 },
]